Source/Evolve/WarpXEvolve.cpp

```diff
@@ -96,7 +96,9 @@ WarpX::Evolve (int numsteps)
             FillBoundaryE(guard_cells.ng_alloc_EB, guard_cells.ng_Extra);
             FillBoundaryB(guard_cells.ng_alloc_EB, guard_cells.ng_Extra);
 #ifdef WARPX_MAG_LLG
-            FillBoundaryM(guard_cells.ng_alloc_EB, guard_cells.ng_Extra);
+            if (WarpX::em_solver_medium == MediumForEM::Macroscopic) {
+                FillBoundaryM(guard_cells.ng_alloc_EB, guard_cells.ng_Extra);
+            }
```

Comment:
Wondering if you should remove the "if" test in the possible case there is a non-macroscopic version of evolveM someday (and you don't want to have to remember to undo all these "if" tests).  You have aborts elsewhere preventing non-macroscopic evolveM from running anyway, so calling FillBoundaryM no matter what makes the code cleaner without changing the functionality.